### PR TITLE
add azure service connector sample in metadata

### PR DESCRIPTION
**Pull request by ChenTanyi**
I'm trying the add azure-service-linker sample in https://github.com/Azure/azure-sdk-for-java/pull/28333,
But it is blocked by the metadata script, so add metadata here.

### CODE REVIEW

**Comment by ChenTanyi** [comment]
As product name is service connector, change name to service connector https://docs.microsoft.com/azure/service-connector
@weshaggard please help to review it.

File: eng/common/scripts/Test-SampleMetadata.ps1 [hunk 1]
```diff
@@ -269,6 +269,7 @@ begin {
         "azure-search",
         "azure-security-center",
         "azure-sentinel",
+        "azure-service-connector",
```
Comment (weshaggard) [inline eng/common/scripts/Test-SampleMetadata.ps1:272]: Is that the correct one or should it be azure-service-linker?

Comment (weshaggard) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: I don't see either at https://review.docs.microsoft.com/en-us/help/contribute/metadata-taxonomies?branch=main#product. can you please find one that is on that list? Also can you update the link in the comments to that same link (i.e. replace master with main).

Comment (ChenTanyi) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: I don't know how to add to the list. Is it any docs about it?
The product name is Service Connector as I mentioned above, I'm not sure it should be azure-service-linker or azure-service-connector. What is the guidance?

Comment (weshaggard) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: I'm less familiar with these doc slugs. @heaths @sima-zhu @danieljurek any recommendations about these service slugs?

Comment (ChenTanyi) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: Thanks wes, maybe I'll add "azure" only in product for sample if this metadata is hard to add

Comment (weshaggard) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: I don't have a lot of context so I cannot provide good guidance here but I think you could try using "azure" as a starting point. 

@scottaddie do you have any suggestions on this front?

Comment (sima-zhu) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: It is usually in a format of "word-word". 
Found `service-connector` in taxonomies. 
Check [here](https://taxonomy.docs.microsoft.com/TaxonomyServiceAdminPage/#/taxonomy/detail/2022-04-27T01:35:49.5155761Z!669b6b65-c764-4717-bbd3-77d5c8065769/ms.service)

Comment (scottaddie) [reply eng/common/scripts/Test-SampleMetadata.ps1:272]: `service-connector` is also my recommendation

**Review by weshaggard — APPROVED** [review]

**Pull request closed without merging**